src/main/java/mil/dds/anet/config/AnetConfiguration.java

```diff
@@ -234,4 +319,24 @@ public void setSslTrust(String sslTrust) {
     }
   }
 
+  @SuppressWarnings("unchecked")
+  public String loadVersion() {
+    try (final InputStream inputStream =
+        AnetConfiguration.class.getResourceAsStream("/version.properties")) {
+      final Map<String, Object> version = yamlMapper.readValue(inputStream, Map.class);
+      this.setVersion((String) version.get("projectVersion"));
+    } catch (IOException e) {
+      logger.error(AnetConstants.VERSION_INFORMATION_ERROR_MESSAGE, e);
+    }
+    return this.getVersion();
+  }
+
+  public String getVersion() {
+    return version == null ? "-" : version;
+  }
+
+  public void setVersion(String version) {
+    this.version = version;
+  }
```

Comment:
I'd change these three methods to something like:
```
  private final Object versionLock = new Object();

  public String getVersion() {
    synchronized (versionLock) {
      if (version == null) {
        try (final InputStream inputStream =
            this.getClass().getResourceAsStream("/version.properties")) {
          @SuppressWarnings("unchecked")
          final Map<String, String> props = yamlMapper.readValue(inputStream, Map.class);
          version = props.getOrDefault("projectVersion", "unknown");
        } catch (IOException e) {
          logger.error(AnetConstants.VERSION_INFORMATION_ERROR_MESSAGE, e);
          version = "!error!";
        }
      }
      return version;
    }
  }
```
(the `synchronized` may be overkill)

Note: please put markers like `@SuppressWarnings("unchecked")` as much as possible on the relevant assignment instead of on the whole method (which may lead to other suppressions that you then don't notice).